terminator/__main__.py

```diff
@@ -6,22 +6,29 @@
 
 
 def pause():
-    time.sleep(arguments.polling_interval)
+    time.sleep(arguments.refresh_job_interval)
 
 
 def loop():
-    previous_jobs = []
+    jobs = []
+    refresh_counter = 0
 
     while True:
-        if not previous_jobs:
-            display.clear_screen()
 
-        current_jobs = jenkins.get_jobs()
+        if refresh_counter >= arguments.refresh_view_frequency:
+            job_names = jenkins.get_job_names()
+            jobs = jenkins.get_jobs(job_names)
+            refresh_counter = 0
+        elif not jobs:
+            display.clear_screen()
+            job_names = jenkins.get_job_names()
+            jobs = jenkins.get_jobs(job_names)
+        else:
+            jobs = jenkins.refresh_jobs(jobs)
 
-        if previous_jobs != current_jobs:
```

Comment:
@Discobeard without this optimisation mechanism anymore, the screen will be repainted every 10 seconds. However, this is the exact optimisation which caused issue #1. The only reason I had the optimisation was to avoid a screen flicker every 10 seconds even though there haven't been any changes to the jobs themselves. I will have a think about this.